ui/components/MesheryPerformanceComponent.js

```diff
@@ -117,8 +117,15 @@ class MesheryPerformanceComponent extends React.Component {
 
   handleChange = (name) => (event) => {
     if (name === 'url' && event.target.value !== '') {
-      this.setState({ urlError: false });
+      let urlPattern = event.target.value;
+      let val = urlPattern.match(/(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/g);
+      if ( !val )
+        this.setState({ urlError: true });
```

Comment:
You would want to set `blockRunTest:true` when URL is invalid

Reply:
Oh yes, but will have to create a new state as using that one would also cause the circular wheel to come up on the button :')

Reply:
the button will be disabled in `BlockRunTest:True` so circular wheel won't come.

Reply:
And that's exactly when the circular wheel comes up.. When the blockruntest is true

Reply:
Instead of adding new state, I think just adding the check with urlError state is sufficient.

Reply:
I tried it on local, UrlError doesn't disable the button rather the server is pinged/

Reply:
Aree, I meant I will be adding that. It is not there currently :').....But, that also is not good to use as initially the urlError will be false and the button won't be disabled. But, we want the button to be disabled until the URL is fine. So, I will add a new state and use that to check it.